### PR TITLE
Add nodejs profiling support

### PR DESCRIPTION
**Pull request by sladyn98**
## Related issue (if exists)

https://github.com/web-infra-dev/rspack/issues/3247

## Summary

This TypeScript script allows you to profile a Node.js-based rspack binary. It checks if the --profile command-line argument is provided. If it is, an Inspector session is initiated and the V8 profiler is started before executing the rspack binary. When the rspack process completes, the profiler is stopped, and the collected CPU profiling data is written to a .cpuprofile file. This script is helpful for identifying performance issues in the rspack binary by providing detailed CPU profiling information.

### <samp>🤖 Generated by Copilot at 01e42eb</samp>

Add a new file `profiler.ts` that enables CPU profiling for rspack. The file uses the inspector API and a child process to run rspack with the `--cpu-prof` flag and save the profile data.

<details open=true>
  <summary><h2>Walkthrough</h2></summary>

### <samp>🤖 Generated by Copilot at 01e42eb</samp>

* Add a profiling option to rspack for performance analysis and optimization

</details>

### CODE REVIEW

File: packages/rspack/src/profiler.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import { performance } from 'perf_hooks';
+import { spawn, ChildProcess } from 'child_process';
+import { Session } from 'inspector';
+import fs from 'fs';
+
+const profileIndex = process.argv.indexOf('--profile');
+
+function start(): ChildProcess {
+    const rspackProcess = spawn('node', ['--cpu-prof', './path-to-rspack/rspack', '-c', 'rspack.config.js'], {
```
Comment (sladyn98) [inline packages/rspack/src/profiler.ts:9]: I would need to get the rspack binary path, do I need to grab this from the config file

**Review by sladyn98 — COMMENTED** [review]
Do I create a new command in rpsack that would invoke this script

File: packages/rspack/src/profiler.ts (hunk 1 above)
Comment (sladyn98) [reply packages/rspack/src/profiler.ts:9]: cc @hardfist

Comment (hardfist) [reply packages/rspack/src/profiler.ts:9]: you don't need to spawn a process here, profile session can be trigger by node api like https://github.com/vitejs/vite/blob/92b7fbb1c9526d68a05e370cc246c75d9e1a6e5e/packages/vite/bin/vite.js#L53

**Comment by hardfist** [comment]
closed in favor of #3937

**Pull request closed without merging**